challenges/2022-10-11-blast-from-the-past/solutions/javascript/justinmemphis/justinmemphis.js

```diff
@@ -0,0 +1,47 @@
+// # Find Three Largest Numbers
+//
+// Write a function that takes in an array of at least three integers and, without
+// sorting the input array, returns a sorted array of the three largest integers in
+// the input array. The function should return duplicate integers if necessary.
+//
+// ## Business Rules/Errata
+//
+// - The input array should have at least three integers. If it does not, you should
+//   return a null value.
+// - You may not sort the input array
+// - The function should handle duplicate integers; for example `[10, 5, 9, 10, 12]`
+//   should return `[10, 10, 12]`
+// - Constant space -> you will return a new array of 3 integers, and this will be the
+//   only new data structure you create.
+// - Linear time -> you should solve this problem by only passing through the array a
+//   single time.
+
+function findThreeLargestNumbers(arr) {
+  // type your solution here
+	var bigThree = [Number.NEGATIVE_INFINITY,Number.NEGATIVE_INFINITY,Number.NEGATIVE_INFINITY];
+        if (arr.length < 3) {
+                return null;
+        } else {
```

Comment:
Hey @Justinmemphis thanks for the PR. I have some feedback. You don't actually need to wrap the rest of the function in an `else` block. Because there is a `return` statement in the conditional `if (arr.length < 3) {` the function will return if that condition is met. This is essentially an "early exit" conditional, so if it's not met, the rest of the function will execute anyway (which is what is happening in your else block). It's not incorrect, but it is considered to be an anti-pattern. Basically, all of this looks good, but I'd like you to refactor everything in the `else` block to just be in line, so it should look like:
```
function findThreeLargestNumbers(arr) {
        // type your solution here
	var bigThree = [Number.NEGATIVE_INFINITY,Number.NEGATIVE_INFINITY,Number.NEGATIVE_INFINITY];
        if (arr.length < 3) {
                return null;
        }
        for (var i = 0; i < arr.length; i++) {
        ...
        }
        return bigThree;
   }
   ```

Reply:
Dennis, thank you for the feedback.  See new commit I just posted to resolve this.